Parsely/wp-parsely
# Add filter wp_parsely_post_type for NewsArticle

Fix #221

## Description 

Adding a filter to `NewsArticle` is easy. However, the challenging part is to make sure that if any value different from `NewsArticle` is assigned to `@type`, it would be handled correctly. 

Therefore, this is what I've done: 

- Declare which types that Parsely supports based on this info https://www.parse.ly/help/integration/jsonld#distinguishing-between-posts-and-pages
- Handle the conversion from JSON-LD types to respective Parsely page types
- Add the `wp_parsely_post_type` filter. 
- Validate the new @type and fall back it to `NewsAritcle`
- Add tests for this filter.

Side-effect change: 
- I added a commit to update the correct path for `views` files. See my comment below. 

## Tests 

## Test 1 

1. Add this filter to your site. Instead of `Article`, you can use any type mentioned in https://www.parse.ly/help/integration/jsonld#distinguishing-between-posts-and-pages

```php
		add_filter('wp_parsely_post_type', function() {
			return 'Article';
		});
``` 

2. Visit a post, and check its HTML source.
3. Verify this value `"@type":"Article"` in the HTML source. 

## Test 2 

1. Add this filter

```php
		add_filter('wp_parsely_post_type', function() {
			return 'Wrong_Type';
		});
``` 
2. Visit a post, and check its HTML source.
3. Verify this value `"@type":"NewsArticle"` in the HTML source. 
4. There is an error message in the log regarding this wrong type.

## Update 

_Last update: 2020-May-04 05.00 UTC_
To move this PR forward, I am waiting for 3 changes applied in the `develop` branch: 

- [x] #276 - Fix paths of view files.
- [ ] #272 - Separate test cases
- [ ] Gary's work on "PHPUnit 9 items even while using PHPUnit 7.5" so that I don't need to use the old way to test PHP error.
